Core/clim-core/frames.lisp

```diff
@@ -755,46 +756,60 @@ documentation produced by presentations.")
   (setf (slot-value pane 'name) name)
   pane)
 
-(defun do-pane-creation-form (name form)
-  (cond
-    ;; Single form which is a function call
-    ((and (= (length form) 1)
-          (listp (first form)))
-     `(coerce-pane-name ,(first form) ',name))
-    ;; Standard pane denoted by a keyword (i.e `:application')
-    ((keywordp (first form))
-     (case (first form)
-       (:application `(make-clim-application-pane
-                       :name ',name ,@(cdr form)))
-       (:interactor `(make-clim-interactor-pane
-                      :name ',name ,@(cdr form)))
-       (:pointer-documentation `(make-clim-pointer-documentation-pane
-                                 :name ',name ,@(cdr form)))
-       (:command-menu `(make-clim-command-menu-pane
-                                :name ',name ,@(cdr form)))
-       (otherwise `(make-pane ,(first form) :name ',name ,@(cdr form)))))
-    ;; Non-standard pane designator fed to the `make-pane'
-    (t `(make-pane ',(first form) :name ',name ,@(cdr form)))))
+(defun %generic-make-or-reinitialize-pane
+    (panes-for-layout abstract-class-name name &rest initargs)
+  (if-let ((pane (alexandria:assoc-value
+                  panes-for-layout name :test #'eq)))
+    (progn
+      (when-let ((parent (sheet-parent pane)))
+        (sheet-disown-child parent pane))
+      (apply #'reinitialize-instance pane initargs))
+    (apply #'make-pane abstract-class-name :name name initargs)))
```

Comment:
is there a reason why this function body is not inlined in generic-make-or-reinitialize-pane in flet later?

Reply:
It is a significant chunk that would be repeated for each pane in the expansion. I see multiple arguments in favor of having a separate function:

* Less code in the expander function
* Less code in the expansion
  * Easier to debug
  * Quicker compilation
* (small advantage) Changes to the function can take effect without re-expanding all `define-application-frame` forms